cargo-workspaces/src/publish.rs

```diff
@@ -43,7 +47,11 @@ pub struct Publish {
 impl Publish {
     pub fn run(self, metadata: Metadata) -> Result {
         let pkgs = if !self.from_git {
-            self.version
+            let mut version = self.version;
+            if self.dry_run {
+                version.git.no_git_push = true;
```

Comment:
Instead of doing this, can we mock the git execution to print to output instead of performing git action?